sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FluxByteBufferContent.java

```diff
@@ -89,7 +100,36 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
 
     @Override
     public boolean isReplayable() {
-        return false;
+        return isReplayable;
+    }
+
+    @Override
+    public BinaryDataContent toReplayableContent() {
+        Flux<ByteBuffer> bufferedFlux = content
+            .map(buffer -> {
+                if (buffer.isDirect()) {
+                    // deep copy direct buffers, they might be pooled memory or mapped file.
+                    ByteBuffer copy = ByteBuffer.allocate(buffer.remaining());
+                    copy.put(buffer);
+                    copy.flip();
+                    return copy;
+                } else {
+                    // these are heap buffers, safe to retain.
+                    return buffer;
+                }
+            })
+            // collectList() uses ArrayList.
+            .collect(LinkedList::new, (BiConsumer<LinkedList<ByteBuffer>, ByteBuffer>) LinkedList::add)
```

Comment:
LinkedList instead of ArrayList given the number of elements is undetermined?

Reply:
Yeah, the concerns are:
- resizing of arraylist
- capacity limit (one can hit array limit with 4GB of 2byte buffers, which is doable on available hardware today).
- we don't need random access.